xserver/playbook.yaml

```diff
@@ -3,74 +3,76 @@
+    # lagacy to remove xterm to prevent terminal access
+    - name: Uninstall xterm
+      apt: 
+        name: xterm
+        state: absent
 
 # remove old services
     - name: Systemd for openbox
-      file: path=/etc/systemd/system/openbox.service state=absent
+      file: 
+        path: /etc/systemd/system/openbox.service 
+        state: absent
 
     - name: Systemd for x11vnc
-      file: path=/etc/systemd/system/x11vnc.service state=absent
+      file: 
+        path: /etc/systemd/system/x11vnc.service 
+        state: absent
 
     - name: Systemd for novnc-3000
-      file: path=/etc/systemd/system/novnc-3000.service state=absent
+      file: 
+        path: /etc/systemd/system/novnc-3000.service 
+        state: absent
 
-    - file: path=/etc/init/novnc.conf state=absent
+    - name: Remove upstart
+      file: 
+        path: /etc/init/novnc.conf 
+        state: absent
 # end
 
-    - name: Systemd for novnc
-      copy: src=novnc.service dest=/etc/systemd/system/novnc.service
+    - name: Systemd service for govnc
+      copy:
+        src: novnc.service 
+        dest: /etc/systemd/system/novnc.service
+
+    - name: Systemd socket for govnc
+      copy:
+        src: novnc.socket 
+        dest: /etc/systemd/system/novnc.socket
 
     - name: Systemd for vnc
-      copy: src=vnc.service dest=/etc/systemd/system/vnc.service
+      copy:
+        src: vnc.service 
+        dest: /etc/systemd/system/vnc.service
 
-    - copy: src=codio-xserver.sh dest=/etc/profile.d/codio-xserver.sh
-    - copy: src=rc.xml dest=/etc/xdg/openbox/rc.xml
-    - copy: src=index.html dest=/opt/novnc/noVNC-master/index.html
+    - copy: 
+        src: codio-xserver.sh 
+        dest: /etc/profile.d/codio-xserver.sh
+    - copy:
+        src: rc.xml 
+        dest: /etc/xdg/openbox/rc.xml
 
-    - name: vnc restarted
-      systemd:
-        enabled: yes
-        state: restarted
-        daemon_reload: yes
-        name: vnc
-      ignore_errors: yes
-    - name: novnc restarted
-      systemd:
-        enabled: yes
-        state: restarted
-        daemon_reload: yes
-        name: novnc
-      ignore_errors: yes
+# Ansible bug, old versions not aware of sockets
+    - name: Govnc enabled
+      shell: systemctl enable --now novnc.socket
 
-    - name: Install commentjson
-      pip:
-        name: commentjson
-        executable: pip3
-        state: present
-    - name: Put Virtual Desktop to the preview targets
-      add_preview_to_codio:
-    - name: Restore .codio permissions
-      file:
-        path: /home/codio/workspace/.codio
-        owner: codio
-        group: codio
+    - name: Add Virtual Desktop to Preview menu
+      become_user: codio
+      shell: 
+        cmd: '(cat .codio || echo "{}") | sed -E -e "s:(\s+(//|#)|^\s*(//|#)).*$::; /^$/d" | jq  ". * {\"preview\": {\"Virtual Desktop\": \"https://{%raw%}{{domain3050}}{%endraw%}/\"}}" > .codio_tmp ; mv .codio_tmp .codio ; true'
```

Comment:
The file will be empty if the syntax is incorrect.
Can we add a Virtual Desktop preview there as before?